kotest-runner/kotest-runner-junit5/src/jvmMain/kotlin/io/kotest/runner/junit/platform/gradle/GradleClassMethodRegexTestFilter.kt

```diff
@@ -18,41 +18,58 @@ class GradleClassMethodRegexTestFilter(private val patterns: List<String>) : Tes
-      val (pck, classname, path) = GradleTestPattern.parse(pattern)
+   private fun match(pattern: String, descriptor: Descriptor): Boolean {
+      val (prefixWildcard, pck, classname, path) = GradleTestPattern.parse(pattern)
       return when (descriptor) {
-         is Descriptor.TestDescriptor -> descriptor.parts().take(path.size) == path
+         is Descriptor.TestDescriptor -> when (path) {
+           null -> false
+           else -> descriptor.path(false).value.startsWith(path)
+         }
+
          is Descriptor.SpecDescriptor -> when {
+            pck != null && classname != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains("$pck.$classname") ?: false
             pck != null && classname != null -> descriptor.kclass.qualifiedName == "$pck.$classname"
+            pck != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains(pck) ?: true
             pck != null -> descriptor.kclass.qualifiedName?.startsWith(pck) ?: true
+            classname != null && prefixWildcard -> descriptor.kclass.simpleName?.contains(classname) ?: false
             classname != null -> descriptor.kclass.simpleName == classname
             else -> true
          }
       }
    }
 }
 
-data class GradleTestPattern(val pckage: String?, val classname: String?, val path: List<String>) {
+data class GradleTestPattern(
+   val suffixWildcard: Boolean,
+   val pckage: String?,
+   val classname: String?,
+   val path: String?,
+) {
    companion object {
 
       // if the regex starts with a lower case character, then we assume it is in the format package.Class.testpath
       // otherwise, we assume it is in the format Class.testpath
       // the .testpath is always optional, and at least Class or package must be specified
+      // additionally, patterns can start with a * in which case the pattern matches suffixes
       fun parse(pattern: String): GradleTestPattern {
+         require(pattern.isNotBlank())
+
+         val prefixWildcard = pattern.startsWith("*")
+         val pattern2 = pattern.removePrefix("*.").removePrefix("*")
 
-         val tokens = pattern.split('.')
+         val tokens = pattern2.split('.')
          val classIndex = tokens.indexOfFirst { it.first().isUpperCase() }
 
          // if class is not specified, then we assume the entire string is a package
-         if (classIndex == -1) return GradleTestPattern(pattern, null, emptyList())
+         if (classIndex == -1) return GradleTestPattern(prefixWildcard, pattern2, null, null)
 
          // if the class is the first part, then no package is specified
          val pck = if (classIndex == 0) null else tokens.take(classIndex).joinToString(".")
 
          val pathParts = tokens.drop(classIndex + 1)
-         val path = if (pathParts.isEmpty()) emptyList() else pathParts.joinToString(".").split(Descriptor.TestDelimiter)
+         val path = if (pathParts.isEmpty()) null else pathParts.joinToString(".")
```

Comment:
Isn't this just `pattern2`?

in line 60 we split it, but it seems we are rejoining it here without ever making changes to it.

Reply:
pathPaths it tokens drop 1